datafusion/physical-optimizer/src/enforce_sorting/mod.rs

```diff
@@ -151,10 +165,51 @@ fn update_coalesce_ctx_children(
     };
 }
 
-/// The boolean flag `repartition_sorts` defined in the config indicates
-/// whether we elect to transform [`CoalescePartitionsExec`] + [`SortExec`] cascades
-/// into [`SortExec`] + [`SortPreservingMergeExec`] cascades, which enables us to
-/// perform sorting in parallel.
+/// If `repartition_sorts` is enabled,
```

Comment:
😍

Reply:
This is greaet -- thank you @wiedld 

I am not sure it will be rendered as a doc if it is listed on the `impl` -- perhaps it would be better here:

https://github.com/apache/datafusion/blob/a104661a020b895eb155af12575bafe693b8edaf/datafusion/physical-optimizer/src/enforce_sorting/mod.rs#L129-L128

Reply:
I need to fix up these docs more than this, since the flowcharts refer to the `parallelize_sorts` part of EnforceSorting, and not to everything that this optimization pass does.